<launch>
	<arg name="rate" default="20" />
	<arg name="name_prefix" default="seabee3_driver_base" />
	<arg name="name_postfix" default="" />
	<arg name="simulate" default="false" />
	
	<arg name="name" value="$(arg name_prefix)$(arg name_postfix)" />
	<arg name="pkg" value="seabee3_driver_base" />
	<arg name="node_type" value="$(arg pkg)" />

	<arg name="port" default="/dev/seabee3_pwrboard" />

	<node pkg="$(arg pkg)" type="$(arg node_type)" name="$(arg name)" output="screen" args="_rate:=$(arg rate) _simulate:=$(arg simulate)" >
    <param name="pump_motor_val" type="int" value="30" />
		<param name="port" type="string" value="$(arg port)" />
		<param name="shooter1/trigger_time" type="int" value="100" />
		<param name="shooter1/trigger_value" type="int" value="75" />

		<param name="shooter2/trigger_time" type="int" value="100" />
		<param name="shooter2/trigger_value" type="int" value="-75" />
		
		<param name="dropper1/trigger_time" type="int" value="125" />
		<param name="dropper1/trigger_value" type="int" value="75" />
		
		<param name="dropper2/trigger_time" type="int" value="125" />
		<param name="dropper2/trigger_value" type="int" value="55" />
		
		<!-- ========== Motor/Behavior Tuning ========== -->
		
		<param name="speed_m1_dir" type="double" value="-1.0" />
		<param name="speed_m2_dir" type="double" value="-1.0" />
		
		<param name="strafe_m1_dir" type="double" value="-1.0" />
		<param name="strafe_m2_dir" type="double" value="1.0" />
		
		<param name="depth_m1_dir" type="double" value="-1.0" />
		<param name="depth_m2_dir" type="double" value="-1.0" />
	</node>
	
	<arg name="show_reconfigure" default="$(arg simulate)" />
	<include file="$(find seabee3_launch)/dynamic_reconfigure.launch" >
		<arg name="enable" value="$(arg show_reconfigure)" />
		<arg name="name" value="$(arg name)" />
	</include>
</launch>
